### ISSUE TITLE
Problem with verification logic.

### ISSUE THREAD

**Issue by Michel-Leidson** [anoma/namada#2736]
**Problem with verification logic. Allowed the transaction to be accepted. even with an invalid steward address**.

namadac resign-steward --steward tnam1q8q0g5j4k9ume9lapusf7wz9z3n7t8g4mszfsmtq --force --memo tpknam1qq6unacxdta40q7v7e0cvzaynqzrjc2w0qfwxa7z5uwzys872z09gzg2u6v --node http://localhost:26657

_Looking-up public key of tnam1q8q0g5j4k9ume9lapusf7wz9z3n7t8g4mszfsmtq from the ledger...
Enter your decryption password: 
Transaction added to mempool.
Wrapper transaction hash: D309D283BBC4A6E6EC46AB6400CB5A8567011487ECFBADFA4BA660C813F3A1C6
Inner transaction hash: 47BB154BDCFC394910BC696CCCB5507D63E4AAA7C33BD56068DD7B8C2B5854DD
Wrapper transaction accepted at height 78910. Used 23 gas.
Waiting for inner transaction result...
Transaction was successfully applied at height 78911. Used 2989 gas._

As the steward address is invalid, the expected behavior would be for the transaction to be rejected even if the **--force** flag was used.

**Comment by Rigorously** [comment]
Now you spoiled how it was possible to get the transaction in the Shielded Expedition without actually being a steward! ;)

By the way, `dry-run` and `dry-run-wrapper` do not complain either.

```
Enter your decryption password: 
Dry-run result: Transaction was successfully applied. Used 2989 gas.
```

**Comment by Fraccaman** [comment]
only the client checks that `tnam1q8q0g5j4k9ume9lapusf7wz9z3n7t8g4mszfsmtq` is a steward. This is because adding the check in the transaction will consume more gas. If `tnam1q8q0g5j4k9ume9lapusf7wz9z3n7t8g4mszfsmtq` is not a steward, no changes are made to the storage.

**Comment by Michel-Leidson** [comment]
Thank you for getting back to me. However, this is not the case for this specific transaction with hash D309D283BBC4A6E6EC46AB6400CB5A8567011487ECFBADFA4BA660C813F3A1C6. It is stored as successful. It seems there's a discrepancy in the stored data, raising questions about data integrity and validity. So, how can I ensure that the data I'm presenting is valid if what's stored isn't?

**Comment by Michel-Leidson** [comment]
If we look at other similar transactions, when we receive the rejection message we have this![image](https://github.com/anoma/namada/assets/91626069/9b5ef300-bc6e-4183-8c36-1842080cabef)

**Comment by Michel-Leidson** [comment]
Furthermore, for those participating in the shielded testnet, it becomes a way to circumvent the competition since it's not necessary to meet both conditions, which are creating a proposal and getting it approved. It's only necessary to force the transaction so that it passes successfully.

This differs from similar transactions where rejections occur.

**Comment by Fraccaman** [comment]
> Thank you for getting back to me. However, this is not the case for this specific transaction with hash D309D283BBC4A6E6EC46AB6400CB5A8567011487ECFBADFA4BA660C813F3A1C6. It is stored as successful. It seems there's a discrepancy in the stored data, raising questions about data integrity and validity. So, how can I ensure that the data I'm presenting is valid if what's stored isn't?

all transactions included in a block are "stored" in the chain, the difference is that rejected transactions don't apply their changes to the underlying storage. 
In case where you are sending a `resign steward` transaction without being a steward you are basically making a no-op transaction, since nothing is modifying the storage. Should no-op transaction be rejected? Imo they shouldn't.

This whole discussion makes sense only from a "shielded expedition" point of view, since you are assigned point for completing the task. In no other case using `--force` to evade client side checks or adding extra logic inside the transaction making it more expensive in terms of gas would be ideal. 
I dont think this can considered a verification logic bug (I didn't check if the client side check is there, in that case I agree we should add it).

**Comment by Fraccaman** [comment]
> If other similar transactions are being correctly rejected, it is important to ensure consistency and apply the same treatment to all transactions. This would help maintain data integrity and reliability.

The difference is that you are comparing a transaction with invalid storage changes to a no-op transaction. No-op shouldn't be reject.

**Comment by Fraccaman** [comment]
> Furthermore, for those participating in the shielded testnet, it becomes a way to circumvent the competition since it's not necessary to meet both conditions, which are creating a proposal and getting it approved. It's only necessary to force the transaction so that it passes successfully.
> 
> This differs from similar transactions where rejections occur.

Well, well done you have found a "loop hole"! You earned the point because you tried the and tested the protocol :) 
Btw, lets keep discussion around the shielded expedition to discord :)

**Comment by brentstone** [comment]
This has been partially addressed in #2748, which improves the client logging for this tx (the error message will display even if `--force` is passed, but the `--force` behavior is the same).

**Comment by Michel-Leidson** [comment]
I created a proposal, but encountered this error while trying to execute it:

namada init-proposal --data-path proposal.json --pgf-funding --memo tpknam1qq6unacxdta40q7v7e0cvzaynqzrjc2w0qfwxa7z5uwzys872z09gzg2u6v --node http://localhost:33357
Last committed epoch: 18
Looking-up public key of tnam1q88vn26unjqx2yzerga75jln998ykf3s7gtaagy5 from the ledger...
Enter your decryption password: 
Transaction added to mempool.
Wrapper transaction hash: C1E665C1FD4977A791A9467FB8F8281A5D5C8D7ED253E5B19D05D18C049D4EA9
Inner transaction hash: 06E1C17A078A474987ED322A36CDCF52D59986A9FFB8EBB2E1A913F7BF1C526C
Wrapper transaction accepted at height 73985. Used 32 gas.
Waiting for inner transaction result...
Transaction failed.
Details: {
  "inner_tx": null,
  "info": "Error trying to apply a transaction: Transaction runner error: Failed running wasm with: RuntimeError: unreachable\n    at <unnamed> (<module>[639]:0x5962d)\n    at <unnamed> (<module>[60]:0x5155)\n    at <unnamed> (<module>[95]:0x7d8c)\n    at <unnamed> (<module>[608]:0x562d7)\n    at <unnamed> (<module>[745]:0x5b05c)",
  "log": "",
  "height": 73986,
  "hash": "06E1C17A078A474987ED322A36CDCF52D59986A9FFB8EBB2E1A913F7BF1C526C",
  "code": "WasmRuntimeError",
  "gas_used": "3508"
}


Upon attempting another transaction, --force it this time, the proposal creation succeeded.

TX Hash: A40C2FCF4BCEC57BDCC17DAB50AF09E0A742FF8EF273E137A7462DFF7ECA199C

Let me know if opening a new issue is necessary or if this behavior is expected."

See proposal created and approved.


Proposal ID: 239
      approved with 62360044.961000 yes votes, 18941388.220000 no votes and 8047779.005884 abstention votes, total voting power: 240721232.274479, limit (fraction) of total voting power required for counting: 80240410.758080 (0.333333 3 33333)
